Cilsil/Cil/Parsers/ExceptionParser.cs

```diff
@@ -0,0 +1,304 @@
+            exceptionHandlerNode.Instructions.Add(
+                    new Load(identifier,
+                             expression,
+                             expressionType,
+                             state.CurrentLocation));
+
+            // When exception is encountered, we translate the return output to being a null/none type.
+            var exceptionOwnerExpression = new VarExpression(identifier);
+            var storeValueIntoVariable = new Store(expression,
+                                                   new ConstExpression(new IntRepresentation(0, false, true)),
+                                                   expressionType,
+                                                   state.CurrentLocation);
+            exceptionHandlerNode.Instructions.Add(storeValueIntoVariable);
+
+            // Construct an instruction to unwrap exception from returned variable. For example: 
+            // n${i}=_fun___unwrap_exception(n${i-1}:last var type*)
+            CreateExceptionCall(state,
+                                exceptionOwnerExpression,
+                                expressionType,
+                                out var exceptionType,
+                                out var returnVariable,
+                                out _,
+                                out var exceptionCall);
+            exceptionHandlerNode.Instructions.Add(exceptionCall);
+            RegisterNode(state, exceptionHandlerNode, true);
+
+            // The variable e allocated when an exception is encountered, 
+            // i.e. "try { ... } catch(IOException e") { ... }
+            var catchVariable = new LvarExpression(
+                                new LocalVariable(Identifier.CatchIdentifier,
+                                                    state.Method));
+            // Construct a exception catch block when unwrapped exception type is not "System.Object".
+            // Since catch block will always have instantiated exception type.
+            if (exceptionType != state.Method.Module.TypeSystem.Object)
+            {
+
+                state.PushInstruction(instruction, exceptionHandlerNode);
+                (instruction, _) = state.PopInstruction();
+
+                /* Create exception allocation call and branch for handling exception. 
+                If exception allocation is True, store allocated exception expression as caught 
+                exception varaible. For example:
+
+                node2 preds:1 succs:4 exn: Conditional exception handler
+                n$49=_fun___instanceof(n$48:java.io.IOException*,sizeof(t=java.io.IOException):void)
+                PRUNE(n$49, true);
+                *&amp;CatchVar65:void=n$48
+
+                node3 preds:1 succs:5 exn: Conditional exception handler -- Connects to outter/finally block code directly
+                n$49=_fun___instanceof(n$48:java.io.IOException*,sizeof(t=java.io.IOException):void)
```

Comment:
![image](https://user-images.githubusercontent.com/38902361/123918669-f2cf4d00-d938-11eb-8296-71754ddf7e69.png)

The output of instanceof (n$6 here) should be a boolean, not an Exception object.

Reply:
Good catch! I will fix that